### DISCUSSION TITLE
Niko switch long press not detected

### DISCUSSION THREAD

**Discussion by Ignaatious** [Koenkk/zigbee2mqtt#7036] [Devices]
I managed to add the Niko switch via the zigbee2mqtt add-on. I am not able to see the short press and long press. I clicked the actual button but still, it did not get added to entities 

![image](https://user-images.githubusercontent.com/6961554/114283843-4756e080-9a4c-11eb-9986-5b3071f0869a.png)

https://www.zigbee2mqtt.io/devices/170-33505.html

Also, is it not possible to trigger the button click action via home assistant automation?

**Comment by wb441** [comment]
Hi! I also have 2 of this device. Zigbee2mqtt definitely recognizes these commands. 
However, I use them in NodeRed and not HA. Therefore, I believe it should be checked in the definitions for HA if the entries are there as well. https://www.zigbee2mqtt.io/devices/GreenPower_On_Off_Switch.html#manual-home-assistant-configuration

The first step would be to check your zigbee2mqtt log when you push the long press. Then you will see what to look for. Next, try to implement it in the HA configuration until it shows in the HA log.

Good luck :)
